algorithms/perf_test/test_inclusive_scan.cpp

```diff
@@ -0,0 +1,143 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdio>
+#include <cstdint>
+#include <cstring>
+#include <cstdlib>
+#include <random>
+
+#include <benchmark/benchmark.h>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Timer.hpp>
+#include <Kokkos_StdAlgorithms.hpp>
+
+#include "../../core/perf_test/Benchmark_Context.hpp"
+
+namespace KE = Kokkos::Experimental;
+
+using exec_space = Kokkos::DefaultExecutionSpace;
+
+template <class ValueType>
+struct SumFunctor {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const ValueType& a, const ValueType& b) const {
+    return (a + b);
+  }
+};
+
+template <class ValueType>
+struct MaxFunctor {
+  KOKKOS_INLINE_FUNCTION
+  ValueType operator()(const ValueType& a, const ValueType& b) const {
+    if (a > b)
+      return a;
+    else
+      return b;
+  }
+};
+
+// Helper to obtain last element of a view
+template <class T>
+T obtainLastElem(const Kokkos::View<T*, exec_space>& out) {
+  auto h_out = Kokkos::create_mirror_view(out);
+  Kokkos::deep_copy(h_out, out);
+  auto view_size = h_out.extent(0);
+  return h_out(view_size - 1);
+}
+
+template <class T>
+void InclusiveScan(benchmark::State& state,
+                   const Kokkos::View<T*, exec_space>& in,
+                   Kokkos::View<T*, exec_space>& out) {
+  // Default prefix sum scan
+  Kokkos::Timer timer;
+  KE::inclusive_scan("default scan", exec_space(), KE::cbegin(in), KE::cend(in),
```

Comment:
Maybe fence before for good measure to make sure nothing else is included in the time recorded.
```suggestion
  exec_space().fence();
  // Default prefix sum scan
  Kokkos::Timer timer;
  KE::inclusive_scan("default scan", exec_space(), KE::cbegin(in), KE::cend(in),
```